From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "jvde.github" <jvde.github@gmail.com>
Date: Mon, 24 Jan 2022 20:25:15 +0100
Subject: [PATCH] add libusb lib directories in cmake for macos

---
 src/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a2bcb3d..9c921d2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,6 +21,7 @@
 add_library(rtlsdr SHARED librtlsdr.c
   tuner_e4k.c tuner_fc0012.c tuner_fc0013.c tuner_fc2580.c tuner_r82xx.c)
 target_link_libraries(rtlsdr ${LIBUSB_LIBRARIES} ${THREADS_PTHREADS_LIBRARY})
+target_link_directories(rtlsdr PUBLIC ${LIBUSB_LIBRARY_DIRS})
 target_include_directories(rtlsdr PUBLIC
   $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>  # <prefix>/include
-- 
2.45.1

