From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "jvde.github" <jvde.github@gmail.com>
Date: Wed, 26 Jan 2022 19:00:55 +0100
Subject: [PATCH] call interrupt_event_handler in cancel_async

---
 src/librtlsdr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/librtlsdr.c b/src/librtlsdr.c
index b7089a9..2bdda08 100644
--- a/src/librtlsdr.c
+++ b/src/librtlsdr.c
@@ -1976,6 +1976,9 @@ int rtlsdr_cancel_async(rtlsdr_dev_t *dev)
 	if (RTLSDR_RUNNING == dev->async_status) {
 		dev->async_status = RTLSDR_CANCELING;
 		dev->async_cancel = 1;
+
+		libusb_interrupt_event_handler(dev->ctx);
+
 		return 0;
 	}
 
-- 
2.40.0

