From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 14 Mar 2025 10:18:10 -0400
Subject: [PATCH] Ignore transfer error when not in progress, already
 complete/cancelled

---
 src/librtlsdr.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/librtlsdr.c b/src/librtlsdr.c
index a8aae48..1fecc4e 100644
--- a/src/librtlsdr.c
+++ b/src/librtlsdr.c
@@ -2180,8 +2180,14 @@ int rtlsdr_read_async(rtlsdr_dev_t *dev, rtlsdr_read_async_cb_t cb, void *ctx,
 #endif
 					libusb_handle_events_timeout_completed(dev->ctx,
 									       &zerotv, NULL);
-					if (r < 0)
+					if (r < 0) {
+						if (r == LIBUSB_ERROR_NOT_FOUND) {
+							/* ignore transfer not in progress, already
+							* complete, or already cancelled */
+							r = 0;
+						}
 						continue;
+					}
 
 					next_status = RTLSDR_CANCELING;
 				}
-- 
2.47.1

