From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Tue, 29 Aug 2023 13:07:08 -0400
Subject: [PATCH] Set version for conda-forge.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2946ba..f41af7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ include(CMakePackageConfigHelpers)
 # Set the version information here
 set(VERSION_INFO_MAJOR_VERSION 0) # increment major on api compatibility changes
 set(VERSION_INFO_MINOR_VERSION 6) # increment minor on feature-level changes
-set(VERSION_INFO_PATCH_VERSION git) # increment patch for bug fixes and docs
+set(VERSION_INFO_PATCH_VERSION 0.post20230829) # increment patch for bug fixes and docs
 include(Version) # setup version info
 
 ########################################################################
-- 
2.40.0

