From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rtlsdrblog <rtlsdrblog@gmail.com>
Date: Thu, 31 Aug 2023 17:37:22 +1200
Subject: [PATCH] fix rtl_tcp error on windows when hints not initialized to 0

---
 src/rtl_tcp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rtl_tcp.c b/src/rtl_tcp.c
index 772a403..78d625a 100644
--- a/src/rtl_tcp.c
+++ b/src/rtl_tcp.c
@@ -386,7 +386,7 @@ int main(int argc, char **argv)
 	struct sockaddr_storage local, remote;
 	struct addrinfo *ai;
 	struct addrinfo *aiHead;
-	struct addrinfo  hints;
+	struct addrinfo  hints = { 0 };
 	char hostinfo[NI_MAXHOST];
 	char portinfo[NI_MAXSERV];
 	char remhostinfo[NI_MAXHOST];
-- 
2.40.0

